jetpack-start/wp-cli-api.php

```diff
@@ -0,0 +1,221 @@
+			WP_CLI::error( 'Jetpack Start API constants (`WPCOM_VIP_JP_START_API_BASE` and/or `WPCOM_VIP_JP_START_API_TOKEN`) are not defined. Please check the `vip-secrets` file to confirm they\'re there, accessible, and valid.' );
+		}
+
+		if ( defined( 'JETPACK_DEV_DEBUG' ) && true === JETPACK_DEV_DEBUG ) {
+			WP_CLI::warning( 'JETPACK_DEV_DEBUG mode is enabled. Please remove the constant after connection.' );
+		}
+
+		$force_connection = WP_CLI\Utils\get_flag_value( $assoc_args, 'force', false );
+		if ( ! $force_connection && Jetpack::is_active() ) {
+			WP_CLI::error( 'Jetpack is already active; bailing. Run this command with `--force` to bypass this check or disconnect Jetpack before continuing.' );
+		}
+
+		WP_CLI::line( '-- Verifying VIP machine user exists (or creating one, if not)' );
+		$user = $this->maybe_create_user();
+		if ( is_wp_error( $user ) ) {
+			WP_CLI::error( $user->get_error_message() );
+		}
+
+		WP_CLI::line( '-- Fetching keys from Jetpack Start API' );
+		$data = $this->api_fetch_keys( $user );
+		if ( is_wp_error( $data ) ) {
+			$message = $data->get_error_message();
+			if ( self::API_ERROR_EXISTING_SUBSCRIPTION === $message ) {
+				$message = 'There is an existing Jetpack Start subcription for this site. Please disconnect using the `cancel` subcommand and try again.';
+			} elseif ( self::API_ERROR_USER_PERMISSIONS === $message ) {
+				$message = sprintf( 'This site already has an existing Jetpack shadow site but the `%s` WP.com user is either not a member of the site or not an administrator. Please use `add_user_to_blog` on your WP.com sandbox to add the account before continuing.', WPCOM_VIP_MACHINE_USER_LOGIN );
+			}
+			WP_CLI::error( 'Failed to fetch keys from Jetpack Start: ' . $message );
+		}
+
+		WP_CLI::line( '-- Adding keys to site' );
+		$this->install_keys( $data );
+
+		update_option( 'vip_jetpack_start_connected_on', time(), false );
+
+		WP_CLI::success( 'All done! Welcome to Jetpack! ✈️️✈️️✈️️' );
+	}
+
+	private function maybe_create_user() {
+		$user = get_user_by( 'login', WPCOM_VIP_MACHINE_USER_LOGIN );
+		if ( ! $user ) {
+			$cmd = sprintf(
+				'user create --url=%s %s %s --role=%s --display_name=%s --porcelain',
+				escapeshellarg( get_site_url() ),
+				escapeshellarg( WPCOM_VIP_MACHINE_USER_LOGIN ),
+				escapeshellarg( WPCOM_VIP_MACHINE_USER_EMAIL ),
+				escapeshellarg( WPCOM_VIP_MACHINE_USER_ROLE ),
+				escapeshellarg( WPCOM_VIP_MACHINE_USER_NAME )
+			);
+
+			$user_id = WP_CLI::runcommand( $cmd, [
+				'return' => true,
+			] );
+
+			$user = get_userdata( $user_id );
+			if ( ! $user ) {
+				return new WP_Error( 'maybe_create_user-failed', 'Failed to create new user.' );
+			}
+		}
+
+		$user_id = $user->ID;
+		if ( is_multisite() ) {
+			add_user_to_blog( get_current_blog_id(), $user_id, WPCOM_VIP_MACHINE_USER_ROLE );
+
+			if ( ! is_super_admin( $user_id ) ) {
+				grant_super_admin( $user_id );
+			}
+		} else {
+			if ( ! $user->has_cap( WPCOM_VIP_MACHINE_USER_ROLE ) ) {
```

Comment:
I think it would be better to check `$user->roles` than `has_cap()`, see https://core.trac.wordpress.org/ticket/22624#comment:10